### DISCUSSION TITLE
Generating texture with crystal models

### DISCUSSION THREAD

**Discussion by marcoaaz** [mtex-toolbox/mtex#2163] [Ask Anything]
Dear community,

I am trying to get familiar with textural patterns within crystals that depend on the 2D section orientation and location within the grains. It would be useful to know if there is a way to produce cross-sections of the crystal shapes ([models](https://mtex-toolbox.github.io/CrystalShapes.html)). I have seen that you can generate a 'slip system', but can you see the traces of the crystal faces on that plane (orthogonal view to the same plane)? 

Also, could one stack several shapes concentrically to simulate a growing pattern? This would help me match (as I rotate the 2D section plane or the crystal in 3D) what I see in nature (scanning electron microscope cathodoluminescence) with the MTEX crystal models. My goal would be to make a small MatLab interface with several options to play around with the orientation and h-k-l plane distances (e.g., Smorf webpage) that generate different 'zoning' patterns in the viewed section planes. First, artificially, through MTEX, and then, with a simulation that suggest what CL pattern fits best to the real data.

If you could point me in the right direction (software), that would help as well. Ty.

Best wishes,
Marco

**Comment by kilir** [comment]
Hi,
I assume, with "texture" you mean not texture as in the contest of mtex but rather what petrographers and those refer to, right?

One thing you can do is moving the crystal relative to the plane.

```Matlab
% import some hexagonal data
mtexdata titanium

cS = crystalShape.hex(ebsd.CS)
N = Miller(0,-1,1,1,ebsd.CS)

mtexFigure; % required
slices = 0:0.1:1;
col = winter(length(slices));
for p = 1:length(slices)
plotInnerFace(cS+ vector3d(slices(p)*N), N,'faceAlpha',0,'edgeColor',col(p,:))
hold on
% plot(cS+ vector3d(slices(p)*N),'faceAlpha',0) % uncomment if you'd like to see the crystals
end
hold off
```

![test](https://github.com/mtex-toolbox/mtex/assets/11697557/fc263cb7-846c-465c-bc84-96a995ffd4de)


Cheers,

Rüdiger
